jobs/upstream.go

```diff
@@ -89,3 +89,43 @@ var SyncArtifactRecords = &job.Job{
 		return err
 	},
 }
+
+// agentArtifactPath is the local path to the agent artifact store.
+var agentArtifactPath string
+
+// SyncArtifactRecords pushes any unpushed artifact records to the upstream.
+// The actual artifacts aren't pushed by this job.
+var SyncArtifactData = &job.Job{
+	JobHistory: true,
+	Singleton:  false, // this job is safe to run concurrently
+	RunNow:     true,
+	Retention:  job.RetentionHour,
+	Name:       "SyncArtifactData",
+	Schedule:   "@every 30s",
+	Fn: func(ctx job.JobRuntime) error {
+		ctx.History.ResourceType = job.ResourceTypeUpstream
+		ctx.History.ResourceID = api.UpstreamConf.Host
+
+		if agentArtifactPath == "" {
```

Comment:
Why can't we use the default artifact store and then support an inline definition of the connection e.g. `fs://var/artifacts`

Reply:
But we are using the default artifact store. 

An inline definition would not work because saving the artifact on the agent would require a connection id.
We can make it work by going around the foreign key constraint but I don't see the point.

In the agent chart, we can have a default artifacts connection that's of type 'folder' with path '/var/artifacts'